### DISCUSSION TITLE
Protobuf map conversion

### DISCUSSION THREAD

**Discussion by Janaka-Steph** [timostamm/protobuf-ts#286] [General]
Any particular reason protobuf `map` is converted to an object rather than a JS Map, as it is the case with official js plugin?

**Comment by timostamm** [comment]
Hey Stéphane, I assume you mean [ECMAScript Map objects](https://tc39.es/ecma262/multipage/keyed-collections.html#sec-map-objects)? 

They can be pretty nice, but don't seem to offer anything plain objects can't for protobuf. With the spread operator and very straight-forward literals, it appears that plain objects are actually a bit nicer to work with most of the time. If I remember correctly, there's also a performance penalty around maps vs object. So that's how we ended up with object instead of Map. 

BTW, the official js plugin uses its own [Map implementation](https://github.com/protocolbuffers/protobuf/blob/main/js/map.js).

**Reply by Janaka-Steph** [reply]
Thank you very much for your answer!
